src/components/EventSchedule/EventScheduleItem.jsx

```diff
@@ -12,88 +13,73 @@ import Typography from "@material-ui/core/Typography";
 import PropTypes from "prop-types";
 import renderPeriod from "./render-period";
 import useStyles from "./CardStyleConst";
-import {EVENT_DEFAULT_IMAGE} from "../../constants/eventConstants";
 import PhotoService from "../../services/PhotoService";
+import { EVENT_DEFAULT_IMAGE } from "../../constants/eventConstants";
 
 const photoService = new PhotoService();
 
-class EventSchedule extends Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      eventImage: EVENT_DEFAULT_IMAGE,
-    };
-  }
-
-  componentDidMount() {
-    photoService
-      .getPreviewEventPhoto(this.props.item.eventId)
-      .then(eventPreviewImage => {
-        if (eventPreviewImage != null) {
-          this.setState({ eventImage: URL.createObjectURL(eventPreviewImage) });
-        }
-      });
-  }
-
-  componentWillUnmount() {
-    URL.revokeObjectURL(this.state.eventImage);
-  }
-
-  render() {
-    const classes = useStyles;
-    const { id, frequency, periodicity, lastRun, nextRun, title, eventId } =
-      this.props.item;
-    const period = renderPeriod(periodicity, frequency);
-    return (
-      <div className="col-12 col-sm-8 col-md-6 col-xl-4 mt-3">
-        <Card className={classes.card}>
-          <CardHeader
-            title={title}
-            subheader={
-              <Moment format="D MMM YYYY" withTitle>
-                {lastRun}
-              </Moment>
-            }
-          />
-          <CardMedia className={classes.media} title={title}>
-            <Link to={`/eventSchedule/${id}`}>
-              <img
-                src={this.state.eventImage}
-                id={`eventPreviewPhotoImg${eventId}`}
-                alt="EventSchedule"
-                className="w-100"
-              />
-            </Link>
-          </CardMedia>
-          <CardContent>
-            <Typography variant="body2" color="textSecondary" component="p">
-              {period}
-            </Typography>
+const EventSchedule = ({ item }) => {
+  const [eventImage, setEventImage] = useState(EVENT_DEFAULT_IMAGE);
+  useEffect(() => {
+    photoService.getPreviewEventPhoto(item.eventId).then(eventPreviewImage => {
+      if (eventPreviewImage != null) {
+        setEventImage(URL.createObjectURL(eventPreviewImage));
+      }
+    });
+    return () => URL.revokeObjectURL(eventImage);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+  const classes = useStyles;
+  const { id, frequency, periodicity, lastRun, nextRun, title, eventId } = item;
```

Comment:
it would be not bad to make destructing in the top of component

Reply:
This component do not used